src/api/helpers.jl

```diff
@@ -224,6 +224,43 @@ function h5f_get_vfd_handle(file_id, fapl)
     return file_handle[]
 end
 
+function h5f_get_free_sections(file_id, type)
+    nsects = h5f_get_free_sections(file_id, type, 0, C_NULL)
+    if nsects > 0
+        return h5f_get_free_sections(file_id, type, nsects)
+    else
+        return H5F_sect_info_t[]
```

Comment:
It returns a `Vector` of `H5F_sect_info_t` describing the address and size of free sections in the HDF5 file. The first call retrieves the number of free sections. If there are no free sections, it is an error to request 0 free sections with a non-null pointer. Therfore we have to return an empty `Vector` of `H5F_sect_info_t` if `nsects == 0`.

Reply:
Ahh, that makes sense. Thanks for the clarification.

Reply:
I missed the definition below and was looking at the original low-level call.... Is there a reason we are separating out these two methods?  Can we just combine them together again?

Reply:
We could, but I think this more obviously shows that there are two and three argument versions of this method without having to consider a union type of `Nothing` and `Int`. I think this is simpler.

Reply:
How would the union show up?  It looks like it always returns an `H5F_sect_info_t`
It seems like the main use case for `nsects !== h5f_get_free_sections(file_id, type, 0, C_NULL)` is to return fewer entries of the vector.  Is that helpful in some cases? 

I'm looking over the documentation: https://confluence.hdfgroup.org/display/HDF5/H5F_GET_FREE_SECTIONS
It also looks like that for the method on L236. If you pass in nsects+1  where `nsects = h5f_get_free_sections(file_id, type, 0, C_NULL)`  It will only fill in the first `nsects` entries of the `sect_info` vector. Might have to add a check to ensure against this.

Reply:
Yes, it is useful if you would like to avoid allocating new memory and would like to reuse an existing vector rather than creating a new one.

Reply:
or rather in this specific case, if you already know how many free sections there from a previous call.

Reply:
> The third example shows that the first call to H5Fget_free_sections() returns the total number of free-space sections in nsects for the H5FD_MEM_BTREE free-space manager in the file that is associated with fcpl. Even though the second call to H5Fget_free_sections() requests nsects+1sections, the routine retrieves free-space section information in sect_info for only nsects sections. The value in ret is the same as nsects.

I think we need to check `nsects <= ` the nominal number by throwing an argument error or resizing the array. Otherwise  it'll return the undef entries.